### PR TITLE
Improvement to search and compatibility with Tmdb Helper player

### PR DESCRIPTION
**Pull request by arvvoid**
## Please check that your pull request pass this checks:

-  [x] My changes respect the [Kodi add-on development rules](https://kodi.wiki/view/Add-on_rules)
-  [x] My code is both Python 2 and 3 compatible
-  [x] My code follows the code style of this project
-  [x] I made sure there wasn't another Pull Request opened for the same update/change
-  [x] I have successfully tested my changes locally

## Types of changes

-  [x] Bug fix (non-breaking change which fixes an issue)
-  [x] New feature (non-breaking change which adds functionality)
-  [ ] Breaking change (fix or feature that would cause existing functionality to change)
-  [ ] Adding a new region/hbo go version

## Description:

Improve search function for both eu and nordic/Spain handler. This will unlock the ability to create a 
Tmdb Helper player.

### CODE REVIEW

File: hbogolib/handlereu.py [hunk 1]
```diff
@@ -184,7 +184,7 @@ def init_api(self, country):
             'Accept-Encoding': ''
         }
 
-    def setup(self, country):
+    def setup(self, country, forceeng=False):
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Parameters differ from overridden 'setup' method](https://app.codacy.com/manual/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4940571)

**Comment by arvvoid** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- hbogolib/base.py  3
         

Clones removed
==============
+ hbogolib/handlereu.py  -1
+ hbogolib/handlersp.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4940571&bid=16334290)

File: hbogolib/handlereu.py (hunk 1 above)
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [HbogoHandler_eu.setup is too complex (19)](https://app.codacy.com/manual/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4940571)

**Comment by arvvoid** [comment]
Create a player file hbogoeu.json in userdata/addon_data/plugin.video.themoviedb.helper/players/
https://github.com/jurialmunkey/plugin.video.themoviedb.helper/wiki/Player-Function

If Tmdb Helper retrieves content in English use this player:

```
{
    "name"              : "Hbo GO EU",
    "plugin"            : "plugin.video.hbogoeu",
    "priority"          : 500,
    "play_movie"        : [
                            "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH_FORCE_ENG&query={title}",
                            {"title": "{title}", "year": "{year}"}
                          ],
    "play_episode"      : [
                            "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH_FORCE_ENG&query={showname}",
                            {"title": "{showname}"},
                            {"position": "{season}"},
                            {"season": "{season}", "episode": "{episode}"}
                          ],
    "search_movie"      : "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={title}",
    "search_episode"    : "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={showname}"
}
```

If Tmdb Helper retrieves content in another language use this player but Hgo Eu have to retrieve content in the same language:

```
{
    "name"              : "Hbo GO EU",
    "plugin"            : "plugin.video.hbogoeu",
    "priority"          : 500,
    "play_movie"        : [
                            "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={title}",
                            {"title": "{title}", "year": "{year}"}
                          ],
    "play_episode"      : [
                            "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={showname}",
                            {"title": "{showname}"},
                            {"position": "{season}"},
                            {"season": "{season}", "episode": "{episode}"}
                          ],
    "search_movie"      : "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={title}",
    "search_episode"    : "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={showname}"
}
```
Modify priority according to preference

**If you are using Spain/Nordic region modify player this way:**

```
{
    "name"              : "Hbo GO EU",
    "plugin"            : "plugin.video.hbogoeu",
    "priority"          : 500,
    "play_movie"        : [
                            "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH_FORCE_ENG&query={title}",
                            {"title": "{title}"}
                          ],
    "play_episode"      : [
                            "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH_FORCE_ENG&query={showname}",
                            {"title": "{showname}"},
                            {"position": "{season}"},
                            {"season": "{season}", "episode": "{episode}"}
                          ],
    "search_movie"      : "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={title}",
    "search_episode"    : "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={showname}"
}
```

If Tmdb Helper retrieves content in another language use this player but Hgo Eu have to retrieve content in the same language:

```
{
    "name"              : "Hbo GO EU",
    "plugin"            : "plugin.video.hbogoeu",
    "priority"          : 500,
    "play_movie"        : [
                            "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={title}",
                            {"title": "{title}"}
                          ],
    "play_episode"      : [
                            "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={showname}",
                            {"title": "{showname}"},
                            {"position": "{season}"},
                            {"season": "{season}", "episode": "{episode}"}
                          ],
    "search_movie"      : "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={title}",
    "search_episode"    : "plugin://plugin.video.hbogoeu/?mode=4&url=EXTERNAL_SEARCH&query={showname}"
}
```

**Comment by arvvoid** [comment]
Test from branch: https://github.com/arvvoid/plugin.video.hbogoeu/archive/TMDBhelper_player.zip

Has been tested so far only with TMDB helper retrieving content in English EU handler.
Nordic/Spain has not been tested at all yet but should work in theory.

**Comment by arvvoid** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- hbogolib/base.py  3
         

Clones removed
==============
+ hbogolib/handlereu.py  -1
+ hbogolib/handlersp.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4940571&bid=16334290)

File: hbogolib/handlereu.py (hunk 1 above)
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [HbogoHandler_eu.setup is too complex (19)](https://app.codacy.com/manual/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4940571)

**Comment by webmaher** [comment]
I try movies and works OK... I have problem with serie Succession ...  now i make more test and then we will see...

**Comment by webmaher** [comment]
No series to play:

When i try The Outsider S01E01 got this:

2020-02-04 18:10:58.451 T:2548 WARNING: XFILE::CFileFactory::CreateLoader - unsupported protocol(plugin) in plugin://plugin.video.themoviedb.helper/?info=play&widget=True&episode=2&tmdb_id=39185&season=9&type=episode
2020-02-04 18:10:58.451 T:2548   ERROR: InputStream: Error opening, plugin://plugin.video.themoviedb.helper/?info=play&widget=True&episode=2&tmdb_id=39185&season=9&type=episode
2020-02-04 18:11:21.284 T:16332   ERROR: XFILE::CDirectory::GetDirectory - Error getting \
2020-02-04 18:12:53.752 T:16272  NOTICE: [plugin.video.themoviedb.helper]
                                            HTTP Error Code: 404
                                            Request: https://api.trakt.tv//shows/tt8550800/seasons/1/episodes/1/ratings
2020-02-04 18:13:12.494 T:15456  NOTICE: Previous line repeats 1 times.
2020-02-04 18:13:12.494 T:15456  NOTICE: Stopping player



When i try Avenue 5 S01E01 got this:


2020-02-04 18:15:02.172 T:2568 WARNING: XFILE::CFileFactory::CreateLoader - unsupported protocol(plugin) in plugin://plugin.video.themoviedb.helper/?info=play&widget=True&episode=2&tmdb_id=39185&season=9&type=episode
2020-02-04 18:15:02.172 T:2568   ERROR: InputStream: Error opening, plugin://plugin.video.themoviedb.helper/?info=play&widget=True&episode=2&tmdb_id=39185&season=9&type=episode
2020-02-04 18:17:20.552 T:10308   ERROR: XFILE::CDirectory::GetDirectory - Error getting \
2020-02-04 18:18:25.599 T:15948  NOTICE: Stopping player


In both cases i go to TMD Helper - TV Shows - Trending This Week ...

**Comment by arvvoid** [comment]
@webmaher when reporting an issue please always provide full debug logs, partial logs are not useful. Anyhow the metadata returned on season listing in HBO go is wrong that's why it is failing so I modified the player with the new player it works for both movies and series.

Update the TMDB Helper player: https://github.com/arvvoid/plugin.video.hbogoeu/pull/112#issuecomment-581810278

**Comment by arvvoid** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- hbogolib/base.py  3
         

Clones removed
==============
+ hbogolib/handlereu.py  -1
+ hbogolib/handlersp.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4940571&bid=16334290)

**Comment by webmaher** [comment]
Thanks, great work, well done... i trying seven series and all work perfect on my W10 pc... now i go testing on my android tv....
Thanks again :-)

**Comment by webmaher** [comment]
Everything is just perfect, big thanks for now :-)

**Pull request merged**